avocado/plugins/tap.py

```diff
@@ -89,6 +89,9 @@ def pre_tests(self, job):
         if tests > 0:
             self.__write("1..%d", tests)
 
+    def start_tests(self, result):
```

Comment:
You should remove the `self._write("1..%d", tests)` from `pre_tests` and move it here with the proper number of tests...